[Chia Min Jiun] iP

minjiunn opened the pull request:
# ZELK :frowning:
(was previously Duke)

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

Zelk frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/minjiunn/ip).
2. double-click it.
3. add your tasks.
4. and let it manage your tasks for you! :relaxed:
5. edit and delete tasks as you go.

and it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Finding a task
- [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public void run() {
    ui.showWelcome();
    boolean isExit = false;
    while (!isExit) {
        try {
            String nextCommand = ui.readCommand();
            Command c = Parser.parse(nextCommand);
            c.execute(tasks, ui, storage);
            isExit = c.isExit();
        } catch (DukeException e) {
            ui.showError(e.getMessage());
        }
    }
    ui.showExit();
}
```

Commit:
Add Gradle support

minjiunn committed:
Level-1

minjiunn committed:
Updated Level-1

minjiunn committed:
implemented Level-2

minjiunn committed:
Implemented Level-3

minjiunn committed:
Implemented Level-4

minjiunn committed:
implemented Level-5

minjiunn committed:
Implemented Level-6

minjiunn committed:
Automated Text UI Testing

minjiunn committed:
Implemented Level 7

minjiunn committed:
Implement Level 8

minjiunn committed:
Merge branch 'branch-Level-7'

minjiunn committed:
Merge branch 'branch-Level-8'

zbz-lvlv commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,219 @@
+    private static String indent = "        ";
+    private static String divider = " ___________________________________________________________________";
+    private enum TaskTypes {TODO, DEADLINE, EVENT, TASK}
+    private static final String DUKE_FILEPATH = "data/duke.txt";
```
Good job on following the convention for constants here!

zbz-lvlv commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task{
```
You're missing JavaDocs in your codebase. They're required for our submissions.

KJunWei replied:
Could perhaps include Javadoc comments for the classes and methods?

zbz-lvlv commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,219 @@
+                    continue;
+                }
+
+                System.out.println(indent + "new task added: " + tasks.get(tasks.size() - 1)
```
Good use of identation style here

zbz-lvlv commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,219 @@
+            taskType = TaskTypes.EVENT;
+        }
+        switch (taskType) {
+            case TODO:
```
The case and switch should be at the same indentation level.

KJunWei replied:
The case and switch should have the same amount of indentation!

zbz-lvlv reviewed (commented):
Overall good adherence to style. There are some details you'll need to be careful, as they deviate from the default setting in intellij.

zbz-lvlv commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,219 @@
+                System.out.println(indent + "You now have " + tasks.size() + " total tasks in your list \n"
+                        + divider);
+            } else {
+                if (input.contains("todo") || input.contains("deadline") || input.contains("event")) {
```
There's too much nesting here, it can affect the readability of your code.

KJunWei commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+public class Event extends Task{
```
Could perhaps add a space between "Task" and "{"!

KJunWei commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+public class Event extends Task{
+
+    protected String on;
```
Great use of OOP principles!

KJunWei commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,219 @@
+                break;
+
+            case EVENT:
+                int eventChar = taskToAdd.indexOf("/");
```
Good job on the use of camelCase!

KJunWei reviewed (commented):
Good job overall!! :)

minjiunn committed:
Implemented More OOP

minjiunn committed:
Implemented packages, fixed bugs

minjiunn committed:
Merge branch 'add-gradle-support' of https://github.com/minjiunn/ip

minjiunn committed:
implemented gradle and added tests

minjiunn committed:
Included JavaDoc for public methods

minjiunn committed:
Update code to follow coding standard

minjiunn committed:
Implemented Level 9

minjiunn committed:
Merge branch 'branch-A-JavaDoc'

minjiunn committed:
Merge branch 'branch-A-CodingStandard'

minjiunn committed:
Merge branch 'branch-Level-9'

minjiunn committed:
Minor changes to Level-9

DarrenCsAcc reviewed (commented):
.

leeianee commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,47 @@
+
+import duke.command.Command;
+
+/** A bot to help you to keep track of your tasks in a to-do list. */
```
Good job with the descriptions for all the classes, they are really easy to understand

leeianee commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,56 @@
+    public static Command parse(String strToParse) throws DukeException {
+        String[] splitTask = strToParse.split(" ", 2);
+        String keyword = splitTask[0];
+        if (!ADD_COMMANDS.contains(keyword) && !EDIT_COMMANDS.contains(keyword) && !VIEW_COMMANDS.contains(keyword)
```
Perhaps using a switch statement here could be better so that it is more readable

leeianee commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,91 @@
+public class Storage {
+
+    private static String filepath;
+    private static final String DUKE_FILEPATH = "data";
```
I like the use of final variables to define the constant filepaths

leeianee commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,131 @@
+package duke;
+
+import duke.task.*;
```
Perhaps importing the classes individually instead of using a wildcard would be better to adhere to the coding standard.

leeianee commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,131 @@
+    /** The list of tasks in the to-do list */
+    private static ArrayList<Task> taskArr;
+
+    public TaskList() {
```
Not really required, but perhaps having JavaDoc comments for all public methods will make it more consistent.

leeianee reviewed (commented):
Good work! Overall the code was very clean and readable, with good OOP.

minjiunn committed:
implemented Level 10

minjiunn committed:
no message

minjiunn committed:
Add Assertions

minjiunn committed:
Improve Code Quality

minjiunn committed:
Merge pull request #2 from minjiunn/branch-A-Assertions

Add Assertions

minjiunn committed:
Merge branch 'master' of https://github.com/minjiunn/ip

minjiunn committed:
Merge branch 'master' into branch-A-CodeQuality

minjiunn committed:
Merge pull request #3 from minjiunn/branch-A-CodeQuality

Improve Code Quality

minjiunn committed:
Add Extension C-Update

The current project supports only basic adding and deleting features of managing tasks in a todo-list.

Expanding on the features improves the capabilities of the bot.

Let's improve the features and add a method for the bot to update a task in the todo-list.

minjiunn committed:
Merge branch 'branch-BCD-Extension'

minjiunn committed:
Improve GUI

minjiunn committed:
no message

minjiunn committed:
Add User Guide

minjiunn committed:
Update README

minjiunn committed:
Update creation of Jar file settings

minjiunn committed:
Fix Gui files

minjiunn committed:
update README
